src/Libraries/Microsoft.Extensions.AI/TextToImage/LoggingTextToImageClient.cs

```diff
@@ -0,0 +1,165 @@
+                    LogCompletedSensitive(nameof(GenerateImagesAsync), AsJson(response));
+                }
+                else
+                {
+                    LogCompleted(nameof(GenerateImagesAsync));
+                }
+            }
+
+            return response;
+        }
+        catch (OperationCanceledException)
+        {
+            LogInvocationCanceled(nameof(GenerateImagesAsync));
+            throw;
+        }
+        catch (Exception ex)
+        {
+            LogInvocationFailed(nameof(GenerateImagesAsync), ex);
+            throw;
+        }
+    }
+
+    /// <inheritdoc/>
+    public override async Task<TextToImageResponse> GenerateEditImageAsync(
+        Stream originalImage, string originalImageFileName, string prompt, TextToImageOptions? options, CancellationToken cancellationToken = default)
+    {
+        if (_logger.IsEnabled(LogLevel.Debug))
+        {
+            if (_logger.IsEnabled(LogLevel.Trace))
+            {
+                LogInvokedSensitive(nameof(GenerateEditImageAsync), prompt, AsJson(options), AsJson(this.GetService<TextToImageClientMetadata>()));
+            }
+            else
+            {
+                LogInvoked(nameof(GenerateEditImageAsync));
+            }
+        }
+
+        try
+        {
+            var response = await base.GenerateEditImageAsync(originalImage, originalImageFileName, prompt, options, cancellationToken);
+
+            if (_logger.IsEnabled(LogLevel.Debug))
+            {
+                if (_logger.IsEnabled(LogLevel.Trace))
+                {
+                    LogCompletedSensitive(nameof(GenerateEditImageAsync), AsJson(response));
+                }
+                else
+                {
+                    LogCompleted(nameof(GenerateEditImageAsync));
+                }
+            }
+
+            return response;
+        }
+        catch (OperationCanceledException)
+        {
+            LogInvocationCanceled(nameof(GenerateEditImageAsync));
+            throw;
+        }
+        catch (Exception ex)
+        {
+            LogInvocationFailed(nameof(GenerateEditImageAsync), ex);
+            throw;
+        }
+    }
+
+    private string AsJson<T>(T value) => LoggingHelpers.AsJson(value, _jsonSerializerOptions);
+
+    [LoggerMessage(LogLevel.Debug, "{MethodName} invoked.")]
+    private partial void LogInvoked(string methodName);
+
+    [LoggerMessage(LogLevel.Trace, "{MethodName} invoked: Prompt: {Prompt}. Options: {TextToImageOptions}. Metadata: {TextToImageClientMetadata}.")]
+    private partial void LogInvokedSensitive(string methodName, string prompt, string textToImageOptions, string textToImageClientMetadata);
+
+    [LoggerMessage(LogLevel.Debug, "{MethodName} completed.")]
+    private partial void LogCompleted(string methodName);
+
+    [LoggerMessage(LogLevel.Trace, "{MethodName} completed: {TextToImageResponse}.")]
+    private partial void LogCompletedSensitive(string methodName, string textToImageResponse);
```

Comment:
Is this going to include all the bytes (as a data URI) for the generated image? That's a whole lot to log. We should probably think more about what we want to do there.

Reply:
Yeah, it looks like it.  The response object serialized as JSON will contain the entire image as base64 string. What do you suggest we do here, any examples for filtering already?

Reply:
I think we probably just don't need this LogCompletedSensitive. It exists elsewhere to log the result data (when in LogLevel.Trace) but here we explicitly don't want to log the result data, at least not when it contains DataContent instead of UriContent. We might want to just condition the call site based on the response we get back.

Reply:
I tried a couple things - one was to create a new response with truncated data, but this feels wrong since it might lead folks to think they were actually getting that.  I went with only calling this if we had a response with no DataContent.